File: LABA5/parserPack/StringParser.java
package parserPack;

import exceptionsPack.EmptyStringException;

public class StringParser {
    public static String parse(String line) throws EmptyStringException {
        StringBuilder builder = new StringBuilder(line);
        do {
            if (builder.lastIndexOf(" ") == builder.length() - 1) {
                builder.deleteCharAt(builder.length() - 1);
            }
            if (builder.indexOf(" ") == 0) {
                builder.deleteCharAt(0);
            }
            if (builder.indexOf("  ") != -1) {
                builder.deleteCharAt(builder.indexOf("  "));
            }
        } while (builder.lastIndexOf(" ") == builder.length() - 1 || builder.indexOf(" ") == 0 || builder.indexOf("  ") != -1);
        if (builder.isEmpty()) throw new EmptyStringException();
        return builder.toString();
    }
}

File: LABA5kt/src/parserPack/StringParser.kt
package parserPack

import exceptionsPack.EmptyStringException

object StringParser {
    /**
     * Parses the input string by removing leading and trailing whitespaces,
     * and collapsing multiple consecutive whitespaces into a single space.
     *
     * @param  line  the input string to be parsed
     * @return       the parsed string
     */
    @Throws(EmptyStringException::class)
    fun parse(line: String): String {
        var line1 = line.trim { it <= ' ' }
        if (line1.isEmpty()) throw EmptyStringException()
        while (line1.contains("  ")) line1 = line1.replace(" {2}".toRegex(), " ")
        return line1
    }
}